### PR TITLE
Event Queue

### PR DESCRIPTION
**Pull request by jakubgiminski**
- [x] EventDispatcher
- [x] Event Listeners
- [x] Event Subscriptions
- [x] Event Subscriptions Provider
- [x] Listener Provider
- [x] Event Queue Item
- [x] Event Queue Item Publisher

### CODE REVIEW

**Review by blake83 — APPROVED** [review]

File: test/Event/EventQueueItemTest.php [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+use Comquer\Event\EventQueueItem;
+use ComquerTest\Fixture\Event\ItemAdded;
+use ComquerTest\Fixture\Event\UpdateShoppingListProjection;
+use PHPUnit\Framework\TestCase;
+
+class EventQueueItemTest extends TestCase
+{
+    /** @test */
+    function getters()
```
Comment (aricl) [inline test/Event/EventQueueItemTest.php]: Is  it particularly valuable to test these getters? In general I have found that testing getters makes classes less maintainable without providing much value

**Review by aricl — APPROVED** [review]

**Review by blake83 — APPROVED** [review]

File: src/Event/Subscription/AggregateEventsSubscription.php [hunk 2]
```diff
@@ -0,0 +1,36 @@
+<?php declare(strict_types=1);
+
+namespace Comquer\Event\Subscription;
+
+
```
Comment (jakubgiminski) [inline src/Event/Subscription/AggregateEventsSubscription.php]: empty line

File: src/Event/Subscription/EventSubscriptionProvider.php [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+        $collection = self::parseAggregateTypesConfig($subscriptions, $collection);
+
+        return $collection;
+    }
+
+    public function getForEvent(Event $event) : self
+    {
+        $filteredSubscriptions = new self();
+
+        foreach ($this as $subscription) {
```
Comment (jakubgiminski) [inline src/Event/Subscription/EventSubscriptionProvider.php]: refactor

File: test/Event/EventSubscriptionProviderTest.php [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+use ComquerTest\Fixture\Event\UpdateUserProjection;
+use ComquerTest\Fixture\Event\UserCreated;
+use PHPUnit\Framework\TestCase;
+
+class EventSubscriptionProviderTest extends TestCase
+{
+    static function getArrayConfig() : array
+    {
+        return [
+            'eventNames' => [
```
Comment (jakubgiminski) [inline test/Event/EventSubscriptionProviderTest.php]: Enforce those two keys always exist and their values are always arrays.

**Pull request merged**